src/ansi-c/ansi_c_convert_type.cpp

```diff
@@ -108,7 +108,7 @@ void ansi_c_convert_typet::read_rec(const typet &type)
     const exprt &as_expr=
       static_cast<const exprt &>(static_cast<const irept &>(type));
     assert(as_expr.operands().size()==1);
-    msc_based=as_expr.op0();
+    msc_based = to_unary_expr(as_expr).op();
```

Comment:
In this case we should get rid of the `assert` just above.

Reply:
done